# Exploring a Real Dataset with Pandas


## Step 1: Load the Dataset

I loaded the Titanic dataset and explored its structure by finding the shape, columns, and data types.

In [100]:
import pandas as pd
df = pd.read_csv("Titanic-Dataset.csv")
print("Shape :",df.shape)
print("Columns :", df.columns)
print("Data type :", df.dtypes)



Shape : (891, 12)
Columns : Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='str')
Data type : PassengerId      int64
Survived         int64
Pclass           int64
Name               str
Sex                str
Age            float64
SibSp            int64
Parch            int64
Ticket             str
Fare           float64
Cabin              str
Embarked           str
dtype: object


## Step 2: Count missing values per column and handle them (fill or drop) with a documented justification.


### Numbers of missing values

In [ ]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

### Fill the missing values

I chose to fill the missing values in the "Cabin" column instead of dropping them because dropping this column would result in losing valuable information from the dataset.

In [115]:
df["Cabin"]=df["Cabin"].fillna("Unknown")

### Verify the results

In [110]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin            0
Embarked         2
dtype: int64

 ### repeat the same steps for the age column

In [111]:
df["Age"] = df1["Age"].fillna(df["Age"].mean())

In [112]:
df.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Cabin          0
Embarked       2
dtype: int64

## Step 3: Filter the Data
Filter the dataset to include only adult passengers (Age > 17).

In [108]:
filteres_age=df[df["Age"] > 17]
print("Number of adult passengers:", len(filteres_age))

Number of adult passengers: 601


### Analysis

The dataset was filtered to include only adult passengers (Age> 17). After filtering, there were 601 adult passengers. This subset can be used to analyze the characteristics of adult passengers, such as their survival rate, passenger class, and ticket fare.

## step 4: GroupBy Analysis

### Average Age by Gender

I used groupby to divide the Titanic passengers based on their gender and calculated the average age for each group.

In [113]:
df.groupby("Sex")["Age"].mean()

Sex
female    28.216730
male      30.505824
Name: Age, dtype: float64

### Survival Rate and Maximum Age by Gender

I used groupby with aggregation to analyze passengers based on gender.
I calculated the maximum age and the average survival rate for each gender group. The survival rate shows the proportion of passengers who survived in each group, where a higher value indicates a greater chance of survival.

In [114]:
df.groupby("Sex").agg({"Age":"max",
"Survived":"mean"})

,Age,Survived
Sex,,
female,63.0,0.742038
male,80.0,0.188908
